### ISSUE TITLE
no-confusing-arrow enable allowParens: true by default

### ISSUE THREAD

**Issue by aladdin-add** [eslint/eslint#11503]
**What rule do you want to change?**
no-confusing-arrow
**Does this change cause the rule to produce more or fewer warnings?**
fewer
**How will the change be implemented? (New option, new default behavior, etc.)?**
enable `allowParens: true` by default
**Please provide some example code that this change will affect:**

```js
var x = a => (1 ? 2 : 3);
```

**What does the rule currently do for this code?**
report an error.
**What will the rule do after it's changed?**
no error reported.
**Are you willing to submit a pull request to implement this change?**
yes.

**Comment by platinumazure** [comment]
Although this will probably result in strictly fewer warnings, I'm adding "breaking" as users would need to change their configurations to preserve the current behavior if that's what they desire.

I'm :+1: for the change (as semver-major).

**Comment by g-plane** [comment]
Really valuable change.

**Comment by not-an-aardvark** [comment]
This proposal was accepted in [yesterday's TSC meeting](https://github.com/eslint/tsc-meetings/blob/137fbe2be55499cd75f28a008900803078d22dfd/notes/2019/2019-03-14.md#no-confusing-arrow-enable-allowparens-true-by-default).
